pkg/oc/admin/migrate/legacyhpa/hpa.go

```diff
@@ -0,0 +1,209 @@
+)
+
+var (
+	defaultMigrations = map[versionKind]versionKind{
+		// legacy oapi group
+		{"v1", "DeploymentConfig"}: {"apps.openshift.io/v1", "DeploymentConfig"},
+
+		// webconsole shenaniganry
+		{"extensions/v1beta1", "DeploymentConfig"}:      {"apps.openshift.io/v1", "DeploymentConfig"},
+		{"extensions/v1beta1", "Deployment"}:            {"apps/v1", "Deployment"},
+		{"extensions/v1beta1", "ReplicaSet"}:            {"apps/v1", "ReplicaSet"},
+		{"extensions/v1beta1", "ReplicationController"}: {"v1", "ReplicationController"},
+	}
+
+	internalMigrateLegacyHPALong = templates.LongDesc(fmt.Sprintf(`
+		Migrate Horizontal Pod Autoscalers to refer to new API groups
+
+		This command locates and updates every Horizontal Pod Autoscaler which refers to a particular
+		group-version-kind to refer to some other, equivalent group-version-kind.
+
+		The following transformations will occur:
+
+%s`, prettyPrintMigrations(defaultMigrations)))
+
+	internalMigrateLegacyHPAExample = templates.Examples(`
+		# Perform a dry-run of updating all objects
+	  %[1]s
+
+	  # To actually perform the update, the confirm flag must be appended
+	  %[1]s --confirm
+
+	  # Migrate a specific group-version-kind to the latest preferred version
+	  %[1]s --initial=extensions/v1beta1.ReplicaSet --confirm
+
+	  # Migrate a specific group-version-kind to a specific group-version-kind
+	  %[1]s --initial=v1.DeploymentConfig --final=apps.openshift.io/v1.DeploymentConfig --confirm`)
+)
+
+func prettyPrintMigrations(versionKinds map[versionKind]versionKind) string {
+	lines := make([]string, 0, len(versionKinds))
+	for initial, final := range versionKinds {
+		line := fmt.Sprintf("		- %s.%s --> %s.%s", initial.APIVersion, initial.Kind, final.APIVersion, final.Kind)
+		lines = append(lines, line)
+	}
+
+	return strings.Join(lines, "\n")
+}
+
+type versionKind struct {
+	APIVersion string
+	Kind       string
+}
+
+func (vk versionKind) GroupVersionKind() (schema.GroupVersionKind, error) {
+	groupVer, err := schema.ParseGroupVersion(vk.APIVersion)
+	if err != nil {
+		return schema.GroupVersionKind{}, err
+	}
+	return groupVer.WithKind(vk.Kind), nil
+}
+
+type MigrateLegacyHPAOptions struct {
+	// maps initial gvks to final gvks in the same format
+	// as HPAs use (CrossVersionObjectReferences) for ease of access.
+	finalVersionKinds map[versionKind]versionKind
+
+	hpaClient autoscalingclient.AutoscalingInterface
+
+	migrate.ResourceOptions
+}
+
+func (o *MigrateLegacyHPAOptions) Bind(c *cobra.Command) {
+	o.ResourceOptions.Bind(c)
+}
+
+// NewCmdMigrateLegacyAPI implements a MigrateLegacyHPA command
+func NewCmdMigrateLegacyHPA(name, fullName string, f *clientcmd.Factory, in io.Reader, out, errout io.Writer) *cobra.Command {
+	options := &MigrateLegacyHPAOptions{
+		ResourceOptions: migrate.ResourceOptions{
+			In:     in,
+			Out:    out,
+			ErrOut: errout,
+
+			AllNamespaces: true,
+			Include:       []string{"horizontalpodautoscalers.autoscaling"},
```

Comment:
You need to add validation that prevents this include from being changed via the options that are binded to this command.

Reply:
done